geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TemporalTiledRasterLayer.scala

```diff
@@ -376,22 +377,29 @@ class TemporalTiledRasterLayer(
   }
 
   def hillshade(
-    sc: SparkContext,
     azimuth: Double,
     altitude: Double,
-    zFactor: Double,
+    zFactorCalculator: ZFactorCalculator,
     band: Int
   ): TiledRasterLayer[SpaceTimeKey] = {
-    val tileLayer = TileLayerRDD(rdd.mapValues(_.band(band)), rdd.metadata)
-
-    implicit val _sc = sc
-
-    val result = tileLayer.hillshade(azimuth, altitude, zFactor)
+    val mt = rdd.metadata.mapTransform
+    val cellSize = rdd.metadata.cellSize
 
-    val multibandRDD: MultibandTileLayerRDD[SpaceTimeKey] =
```

Comment:
I think you can save yourself code duplication here by asking for `K: SpatialComponent` and using `withContextRDD` as the constructor for the `TiledRasterLayer`

Reply:
Oh, that's right. I can clean that up as well.

Reply:
Moving the `hillshade` method to `TiledRasterLayer` results in a `SerializationError`

```
E                   py4j.protocol.Py4JJavaError: An error occurred while calling o50.hillshade.
E                   : org.apache.spark.SparkException: Task not serializable
E                   	at org.apache.spark.util.ClosureCleaner$.ensureSerializable(ClosureCleaner.scala:345)
E                   	at org.apache.spark.util.ClosureCleaner$.org$apache$spark$util$ClosureCleaner$$clean(ClosureCleaner.scala:335)
E                   	at org.apache.spark.util.ClosureCleaner$.clean(ClosureCleaner.scala:159)
E                   	at org.apache.spark.SparkContext.clean(SparkContext.scala:2292)
E                   	at org.apache.spark.rdd.RDD$$anonfun$mapPartitions$1.apply(RDD.scala:797)
E                   	at org.apache.spark.rdd.RDD$$anonfun$mapPartitions$1.apply(RDD.scala:796)
E                   	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
E                   	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
E                   	at org.apache.spark.rdd.RDD.withScope(RDD.scala:363)
E                   	at org.apache.spark.rdd.RDD.mapPartitions(RDD.scala:796)
E                   	at geopyspark.geotrellis.TiledRasterLayer$$anonfun$11.apply(TiledRasterLayer.scala:413)
E                   	at geopyspark.geotrellis.TiledRasterLayer$$anonfun$11.apply(TiledRasterLayer.scala:411)
E                   	at geotrellis.spark.Implicits$WithContextWrapper.withContext(Implicits.scala:70)
E                   	at geopyspark.geotrellis.TiledRasterLayer.hillshade(TiledRasterLayer.scala:411)
E                   	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
E                   	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
E                   	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
E                   	at java.lang.reflect.Method.invoke(Method.java:498)
E                   	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
E                   	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
E                   	at py4j.Gateway.invoke(Gateway.java:282)
E                   	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
E                   	at py4j.commands.CallCommand.execute(CallCommand.java:79)
E                   	at py4j.GatewayConnection.run(GatewayConnection.java:214)
E                   	at java.lang.Thread.run(Thread.java:748)
E                   Caused by: java.io.NotSerializableException: geotrellis.spark.io.json.KeyFormats$SpatialKeyFormat$
E                   Serialization stack:
E                   	- object not serializable (class: geotrellis.spark.io.json.KeyFormats$SpatialKeyFormat$, value: geotrellis.spark.io.json.KeyFormats$SpatialKeyFormat$@583ec3d5)
E                   	- field (class: geopyspark.geotrellis.TiledRasterLayer, name: evidence$2, type: interface spray.json.JsonFormat)
E                   	- object (class geopyspark.geotrellis.SpatialTiledRasterLayer, geopyspark.geotrellis.SpatialTiledRasterLayer@63c8a1de)
E                   	- field (class: geopyspark.geotrellis.TiledRasterLayer$$anonfun$11, name: $outer, type: class geopyspark.geotrellis.TiledRasterLayer)
E                   	- object (class geopyspark.geotrellis.TiledRasterLayer$$anonfun$11, <function1>)
E                   	- field (class: geopyspark.geotrellis.TiledRasterLayer$$anonfun$11$$anonfun$apply$25, name: $outer, type: class geopyspark.geotrellis.TiledRasterLayer$$anonfun$11)
E                   	- object (class geopyspark.geotrellis.TiledRasterLayer$$anonfun$11$$anonfun$apply$25, <function1>)
E                   	at org.apache.spark.serializer.SerializationDebugger$.improveException(SerializationDebugger.scala:40)
E                   	at org.apache.spark.serializer.JavaSerializationStream.writeObject(JavaSerializer.scala:46)
E                   	at org.apache.spark.serializer.JavaSerializerInstance.serialize(JavaSerializer.scala:100)
E                   	at org.apache.spark.util.ClosureCleaner$.ensureSerializable(ClosureCleaner.scala:342)
```

It's not entirely clear to me how this should be resolved, and due to the current time constraints, I'd like to just leave the `hillshade` method in `SpatialTiledRasterLayer` and `TemporalTiledRasterLayer` for now.